smk/QC_0.smk

```diff
@@ -0,0 +1,515 @@
+    result = expand("{wd}/{omics}/1-trimmed/{sample}/{sample}.{group}.read_length.txt",
+                    wd=working_dir,
+                    omics=omics,
+                    sample=ilmn_samples,
+                    group=['1', '2'])
+    return(result)
+
+def qc1_read_length_cutoff_output():
+    result = expand("{wd}/output/1-trimmed/{omics}_cumulative_read_length_cutoff.pdf",
+                    wd = working_dir,
+                    omics = omics)
+    return(result)
+
+def qc1_config_yml_output():
+    result = expand("{wd}/{omics}/QC_2.yaml",
+                    wd = working_dir,
+                    omics = omics)
+    return(result)
+
+def qc0_multiqc_summary_output():
+    result = expand("{wd}/output/1-trimmed/{omics}_multiqc.html",
+                    wd = working_dir,
+                    omics = omics)
+    return(result)
+
+rule all:
+    input:
+        qc1_read_length_output(),
+        qc1_read_length_cutoff_output(),
+        qc1_config_yml_output(),
+        qc0_multiqc_summary_output()
+
+
+###############################################################################################
+# Pre-processing of metaG and metaT data
+# Filter reads by quality
+###############################################################################################
+
+# Get a sorted list of runs for a sample
+# Only for samples in individual folders
+
+def get_runs_for_sample(sample):
+    runs = [sample]
+    if ilmn_samples_organisation == "folder":
+        sample_dir = '{raw_dir}/{sample}'.format(raw_dir=raw_dir, sample=sample)
+        runs = [ f.name.split(ilmn_suffix[0])[0][:-1] for f in os.scandir(sample_dir) if f.is_file() and f.name.endswith(ilmn_suffix[0]) ]
+    #print(runs)
+    return(sorted(runs))
+
+# Get files for a run
+
+def get_raw_reads_for_sample_run(wildcards):
+    from glob import glob
+    prefix = '{raw_dir}/{sample}'.format(raw_dir=raw_dir, sample=wildcards.sample)
+    if ilmn_samples_organisation == "folder":
+        prefix = '{raw_dir}/{sample}/{run}'.format(raw_dir=raw_dir, sample=wildcards.sample, run=wildcards.run)
+    raw_sample_run = {}
+    for i, k in enumerate(['read_fw', 'read_rev']):
+        raw_sample_run[k] = glob("{}*{}".format(prefix, ilmn_suffix[i]))[0]
+    return raw_sample_run
+
+# Get file for infering index
+
+def get_example_to_infer_index(sample):
+    sample_dir = '{raw_dir}/{sample}'.format(raw_dir=raw_dir, sample=sample)
+    runs = [ path.normpath(f) for f in scandir(sample_dir) if f.is_file() and f.name.endswith(ilmn_suffix[0]) ]
+    #print(runs)
+    return(runs[0])
+
+##########
+# FastQC of the initial raw data
+##########
+
+rule initial_fastqc:
+    input:
+        unpack(get_raw_reads_for_sample_run)
+    output:
+        flag="{wd}/{omics}/1-0-qc/{sample}/{sample}-{run}_fastqc.done",
+    params:
+        outdir="{wd}/{omics}/1-0-qc/{sample}"
+    log:
+        "{wd}/logs/{omics}/1-0-qc/{sample}_{run}_qc0_fastqc.log"
+    threads:
+        2
+    conda:
```

Comment:
Is default `mem` enough?

Reply:
It was a local rule, now I made it into a regular rule, with 4 Gb of RAM specified. [49a8c1e](https://github.com/arumugamlab/MIntO/pull/12/commits/49a8c1eda826b38f2c7e2435a111938ea188e84a)